cmake/Modules/FindTPLMKL.cmake

```diff
@@ -7,7 +7,40 @@ IF (CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
   KOKKOSKERNELS_CREATE_IMPORTED_TPL(MKL INTERFACE COMPILE_OPTIONS -mkl LINK_OPTIONS -mkl)
   INCLUDE(FindPackageHandleStandardArgs)
   FIND_PACKAGE_HANDLE_STANDARD_ARGS(MKL DEFAULT_MSG KOKKOSKERNELS_HAS_MKL_ARG)
+ELSEIF(WIN32)
+  SET(BLA_VENDOR Intel10_64lp)
+  FIND_PACKAGE(BLAS REQUIRED)
+  IF (NOT DEFINED ENV{MKLROOT})
+    SET(NO_MKL_ROOT_GIVEN "MKL-NOTFOUND")
+    MESSAGE(WARNING "No MKLROOT environment variable specified - must source mklvars.sh to configure MKL path")
+    FIND_PACKAGE_HANDLE_STANDARD_ARGS(MKL NO_MKL_ROOT_GIVEN)
+  ELSE()
+    KOKKOSKERNELS_CREATE_IMPORTED_TPL(MKL INTERFACE
+      LINK_OPTIONS ${BLAS_LINKER_FLAGS}
+      LINK_LIBRARIES ${BLAS_LIBRARIES}
+    )
+  ENDIF()
 ELSE()
-  MESSAGE(FATAL_ERROR "No support currently for MKL without using the Intel compiler")
+  IF (NOT DEFINED ENV{MKLROOT})
+    SET(NO_MKL_ROOT_GIVEN "MKL-NOTFOUND")
+    MESSAGE(WARNING "No MKLROOT environment variable specified - must source mklvars.sh to configure MKL path")
```

Comment:
@jjwilke what is mklvars.sh?

Reply:
The MKL installation comes with an mklvars.sh that you should source to configure your environment. It's really the only robust way to make sure all the libraries can be found for non-Intel compilers.

Reply:
It comes from the MKL installation, not Kokkos Kernels - in case that wasn't clear. I can change the verbage if that was confusing.

Reply:
Not necessary to change it, thanks for clarifying.